text/_util.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+export function split(input: string) {
+  input = input.trim();
+  if (input.includes(" ")) return input.split(/ +/).filter(Boolean);
+  if (input.includes("-")) return input.split(/-+/).filter(Boolean);
+  if (input.includes("_")) return input.split(/_+/).filter(Boolean);
```

Comment:
```suggestion
  if (/[\s-_]+/.test(input)) return input.split(/[\s-_]+/).filter(Boolean);
```
This should also take care of mixed delimiters if we want to do that.

Reply:
The question is if we want to handle that. If we do, we should also consider handling punctuation as well as different languages chars, such as german umlaute.
I am not sure if we should tackle that or just leave it to the user to prepare the string before applying a case.

Different cases require different transformations, replacing either one or all delimiters. That would mean data loss between them:

```ts
// replacing all delimiters and punctuation
console.log(toCamelCase("I am up-to-date!")); // should be "iAmUpToDate"
console.log(toKebabCase("I am up-to-date!")); // should be "i-am-up-to-date"
console.log(toPascalCase("I am up-to-date!")); // should be "IAmUpToDate"
console.log(toScreamingSnakeCase("I am up-to-date!")); // should be "I_AM_UP_TO_DATE"
console.log(toSnakeCase("I am up-to-date!")); // should be "i_am_up_to_date"

// replacing one delimiter
console.log(toTitleCase("I am up-to-date!")); // should be "I Am Up-to-date!"
console.log(toSentenceCase("I AM UP-TO-DATE!")); // should be "I am up-to-date!"
```
So maybe let `split()` have an `splitAllDelimiters` option  which is true for all cases but `toSentenceCase()` and `toTitleCase()`
and also remove all `/\W/` if splitAllDelimiters is true?

Reply:
No, please, let's keep it simple. Let's use your original implementation. It's good enough for the first pass. Let's discuss options in a separate follow-up issue.